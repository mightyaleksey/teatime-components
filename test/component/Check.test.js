'use strict';

const Check = require('../../component/Check');
const React = require('react');
const renderer = require('react-test-renderer');

test('Check renders correctly', () => {
  const tree = renderer.create(
    <Check
      checked={true}
      className='outer'
      disabled={true}
      id='checks id'
      label='checks label'
      name='check control'/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Check\'s initial size is `m`', () => {
  expect(Check.defaultProps.size).toBe('m');
});
