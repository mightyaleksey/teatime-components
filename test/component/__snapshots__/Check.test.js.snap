// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check renders correctly 1`] = `
<span
  className="wrapper outer"
  onChange={undefined}
>
  <input
    checked={true}
    className="native"
    defaultChecked={undefined}
    disabled={true}
    id="checks id"
    name="check control"
    onBlur={undefined}
    onChange={[Function]}
    onFocus={undefined}
    type="checkbox"
    value={undefined}
  />
  <label
    className="control"
    htmlFor="checks id"
  />
  <label
    className="label"
    htmlFor="checks id"
  >
    checks label
  </label>
</span>
`;
